### PR TITLE
Fix beta issues for React

### LINKED ISSUE #13989: OneToMany id bug in react entity validation form
##### **Overview of the issue**

Default name in AvInput of OneToMany relationship causes wrong object serialization.
I think it happens only in OneToMany relationships with required in the Many side.
image
image

##### **Motivation for or Use Case**

Can't properly create or update entities from frontend.

##### **Reproduce the error**

In JDL Studio >
relationship OneToMany {
    Warehouse{inventory} to Inventory{warehouse required},
}

##### **Related issues**

Didn't find one.

##### **JHipster Version(s)**

Release 7.0.0-beta.1
I have not tried previous versions of JHipster.

##### **JHipster configuration**

`-- generator-jhipster@7.0.0-beta.1

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "blueprints": [],
    "otherModules": [],
    "applicationType": "monolith",
    "jhipsterVersion": "7.0.0-beta.1",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": ["cypress", "gatling", "cucumber"],
    "pages": [],
    "creationTimestamp": 1613098461847,
    "promptValues": {
      "nativeLanguage": "es"
    },
    "serviceDiscoveryType": false,
    "reactive": false,
    "authenticationType": "jwt",
    "serverPort": "8080",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "buildTool": "maven",
    "serverSideOptions": ["websocket:spring-websocket"],
    "websocket": "spring-websocket",
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "react",
    "withAdminUi": true,
    "clientTheme": "litera",
    "clientThemeVariant": "primary",
    "enableTranslation": true,
    "nativeLanguage": "es",
    "clientPackageManager": "npm",
    "languages": ["es", "en"],
    "lastLiquibaseTimestamp": 1613099810000
  }
}
</pre>
</details>

openjdk version "11.0.8" 2020-07-14 LTS
OpenJDK Runtime Environment Corretto-11.0.8.10.1 (build 11.0.8+10-LTS)
OpenJDK 64-Bit Server VM Corretto-11.0.8.10.1 (build 11.0.8+10-LTS, mixed mode)

git version 2.24.1.windows.2

node: v14.15.3

npm: 6.14.9

yeoman: 3.1.1

Docker version 20.10.2, build 2291f61

docker-compose version 1.27.4, build 40524192

##### **Browsers and Operating System**

Windows 10 and Google Chrome.

### LINKED ISSUE #14163: Creation of entity with relationship fails on latest beta
App used is same as https://github.com/jhipster/generator-jhipster/issues/14158

Try creating an entity with a relationship

I'll add in more details later, creating the issue to not loose track of it

@pascalgrimaud @avdev4j if this is something alreday fixed on main let me know. I'll test on main tomorrow and update the ticket otherwise. Too late today

### PR DESCRIPTION
**Pull request by deepu105**
Revert #14121
Fix #13989
Fix #14163 

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Review by qmonmert — APPROVED** [review]

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs [hunk 1]
```diff
@@ -205,21 +192,21 @@ export const <%= entityReactName %>Update = (props: I<%= entityReactName %>Updat
...
           <%_ if (rel.relationshipType === 'many-to-one' && rel.otherEntityName !== 'user')  { _%>
-          , <%= rel.relationshipFieldName %>: <%= rel.otherEntityNamePlural %>.find(<%= rel.relationshipFieldName %> => <%= rel.relationshipFieldName %>.id.toString() === values.<%= rel.relationshipFieldName %>Id.toString())
+          <%= rel.relationshipFieldName %>: <%= rel.otherEntityNamePlural %>.find(it => it.id.toString() === values.<%= rel.relationshipFieldName %>Id.toString()),
           <%_ } _%>
         <%_ }) _%>
+        <%_ if (uniqueRelationFields.has('users')) { _%>
+          <%_ relationships.filter(rel => rel.relationshipType === 'one-to-one' && rel.ownerSide === true && rel.otherEntityName === 'user').forEach(rel => { _%>
+        <%= rel.relationshipFieldName %>: users.find(user => user.id.toString() === values.<%= rel.relationshipFieldName %>Id.toString()),
+          <%_ }) _%>
+        <%_ } _%>
```
Comment (mshima) [inline generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs:208]:
```suggestion
```

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs [hunk 2]
```diff
@@ -205,21 +192,21 @@ export const <%= entityReactName %>Update = (props: I<%= entityReactName %>Updat
...
       const entity = {
         ...<%= entityInstance %>Entity,
-        ...values
+        ...values,
         <%_ manyToManyOwners.forEach(rel => { _%>
-        , <%= rel.relationshipFieldNamePlural %>: mapIdList(values.<%= rel.relationshipFieldNamePlural %>)
+        <%= rel.relationshipFieldNamePlural %>: mapIdList(values.<%= rel.relationshipFieldNamePlural %>),
         <%_ }) _%>
         <%_ relFieldNames.forEach(rel => { _%>
           <%_ if (rel.relationshipType === 'many-to-one' && rel.otherEntityName !== 'user')  { _%>
```
Comment (mshima) [inline generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs:200]:
```suggestion
          <%_ if (rel.relationshipType === 'many-to-one' || (rel.ownerSide && rel.relationshipType === 'one-to-one'))  { _%>
```

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs [hunk 3]
```diff
@@ -205,21 +192,21 @@ export const <%= entityReactName %>Update = (props: I<%= entityReactName %>Updat
...
-          , <%= rel.relationshipFieldName %>: <%= rel.otherEntityNamePlural %>.find(<%= rel.relationshipFieldName %> => <%= rel.relationshipFieldName %>.id.toString() === values.<%= rel.relationshipFieldName %>Id.toString())
+          <%= rel.relationshipFieldName %>: <%= rel.otherEntityNamePlural %>.find(it => it.id.toString() === values.<%= rel.relationshipFieldName %>Id.toString()),
```
Comment (mshima) [inline generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs:201]:
```suggestion
          <%= rel.relationshipFieldName %>: <%= rel.otherEntityNamePlural %>.find(it => it.<%= rel.otherEntity.primaryKey.name %>.toString() === values.<%= rel.relationshipFieldName %>Id.toString()),
```
Custom id support.

**Review by mshima — COMMENTED** [review]
This is a wild guess, but since you have changes condition at line 404, seems to make sense.

**Comment by qmonmert** [comment]
@deepu105 you try this ticket https://github.com/jhipster/generator-jhipster/issues/13989 ?

**Comment by deepu105** [comment]
@mshima not sure about the changes you suggested, need to test it. May be lets do it as a new PR as I want to merge this to fix the bug first

**Comment by deepu105** [comment]
@mshima never mind you are right. I merged your suggestion

**Comment by pascalgrimaud** [comment]
So? Is it ready to be merged?

**Comment by mshima** [comment]
Looks like my changes are not enough.
Maybe @deepu105 or @qmonmert can take a look otherwise we should revert last 2 commits.

**Comment by deepu105** [comment]
Lets just revert the last 2 commits from your suggestion and merge it as it
was working for the issues I encountered. We can optimise later

**Comment by pascalgrimaud** [comment]
Ok, I just restarted the failed build. Feel free to merge it when it's ready :-D
I take care ofher stuff

**Comment by mshima** [comment]
Reverting the 2 commits won’t help.
The failures were happening before them.

**Comment by deepu105** [comment]
Ya I just saw that. Will spend some time tomorrow to look into it

**Pull request merged**